src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/dbscripts/querylogs.sql

```diff
@@ -143,9 +143,13 @@ SELECT
   "WDName"
 FROM "{{baseDatabase}}"."{{#if vars.tableName}}{{vars.tableName}}{{else}}QryLogV{{/if}}"
 {{#if queryLogsVariables.timeRange}}
-WHERE "{{baseDatabase}}"."{{#if vars.tableName}}{{vars.tableName}}{{else}}QryLogV{{/if}}"."StartTime"
+  WHERE "{{baseDatabase}}"."{{#if vars.tableName}}{{vars.tableName}}{{else}}QryLogV{{/if}}"."StartTime"
   BETWEEN TIMESTAMP '{{queryLogsVariables.timeRange.startTimestamp}}' AND TIMESTAMP '{{queryLogsVariables.timeRange.endTimestamp}}'
 {{/if}}
+{{#if queryLogsVariables.users}}
```

Comment:
Handelbars templates don't have best readability and may be hard to maintain. I already feel it with this script

You can move WHERE clause to a helper and define the logic inside the java class ([link](https://handlebarsjs.com/guide/expressions.html#helpers).

It will look something like this:

```
// template
{{#queryLogsWhereClause}}{{/queryLogsWhereClause}}
```

```java
// helper
private void registerHelpers() {
    handlebars.registerHelper(
        "queryLogsWhereClause",
        (Helper<SqlScriptVariables>)
            (variables, options) -> {
              QueryLogsVariables queryLogsVariables = variables.getQueryLogsVariables();
              if (!queryLogsVariables.timeRange().isPresent()
                  && queryLogsVariables.users().isEmpty()) {
                return null;
              }

              StringBuilder whereClause = new StringBuilder().append("WHERE \n");
              queryLogsVariables
                  .timeRange()
                  .ifPresent(
                      timeRange -> {
                        String timeRangeClause =
                            String.format(
                                "%s.%s.\"StartTime\" BETWEEN TIMESTAMP '%s' AND '%s'",
                                wrapInQuotes(variables.getBaseDatabase()),
                                wrapInQuotes(
                                    variables.getVars().getOrDefault("tableName", "QryLogV")),
                                timeRange.getStartTimestamp(),
                                timeRange.getEndTimestamp());
                        whereClause.append(timeRangeClause);
                      });

              if (!variables.users().isEmpty()) {
                queryLogsVariables.timeRange().ifPresent(unused -> whereClause.append("\nAND "));
                whereClause.append(
                    String.format("USERS IN (%s)", Joiner.on(",").join(variables.users())));
              }

              return whereClause.toString();
            });
  }

  private String wrapInQuotes(String input) {
    return String.format("\"%s\"", input);
  }
```

I tested it without `users()` part, but should be working fine

Pros:
* It will eliminate the requirement to have dedicated accessor handlebars `getUsers()`.
* As it can be moved to another class it can be tested in isolation.
* It allows to move other clauses to helpers and reuse the code later.

Reply:
Per offline discussion I think this approach makes sense. Added this but using a slightly different implementation that I think has better readability.